clients/java/client/src/main/java/org/camunda/bpm/client/task/impl/ExternalTaskImpl.java

```diff
@@ -300,8 +301,8 @@ public <T extends TypedValue> T getVariableTyped(String variableName, boolean de
     VariableValue variableValue = receivedVariableMap.get(variableName);
     if (variableValue != null) {
       typedValue = variableValue.getTypedValue(deserializeObjectValues);
-      if(typedValue instanceof DeferredFileValue) {
-        DeferredFileValue deferredFileValue = (DeferredFileValue) typedValue;
+      if(typedValue instanceof DeferredFileValueImpl) {
+        DeferredFileValueImpl deferredFileValue = (DeferredFileValueImpl) typedValue;
         deferredFileValue.setExecutionId(this.executionId);
       }

```

Comment:
❓ Can't we get rid of this code now that we pass the execution id to `VariableValue`?

Reply:
We use this code now to pass the `executionId` to the variable value.